Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/flow/HotDataFlow.kt

```diff
@@ -36,19 +38,24 @@ class HotDataFlow<T : Any>(
         extraBufferCapacity = Int.MAX_VALUE,
         onBufferOverflow = BufferOverflow.SUSPEND
     )
+    private val valueGuard = Mutex()
 
     private val internalFlow = channelFlow {
-        var currentValue = startValueProvider().also {
-            Timber.tag(tag).v("startValue=%s", it)
-            send(it)
+        var currentValue = valueGuard.withLock {
+            startValueProvider().also { send(it) }
         }
+        Timber.tag(tag).v("startValue=%s", currentValue)
 
-        updateActions.collect { updateAction ->
-            currentValue = updateAction(currentValue).also {
-                currentValue = it
-                send(it)
+        updateActions
+            .onCompletion {
+                Timber.tag(tag).v("updateActions resetReplayCache().")
+                updateActions.resetReplayCache()
```

Comment:
When we unsubscribe, and the `HotDataFlow` goes cold, and then resubscribe, we would replay all previous update actions on the new value.

I don't think this affected the current issue, but it's a bug nonetheless.